walk/walk.go

```diff
@@ -397,6 +399,12 @@ func walkDir(root, rel string, eg *errgroup.Group, limitCh chan struct{}, isIgno
 		return err
 	}
 
+	// Sort trie entries once before processing to ensure
```

Comment:
This should be unnecessary as `os.ReadDir` sorts the entries: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/os/dir.go;l=127

Reply:
I was just moving it to be done while iterating over the `ReadDir` instead of done later in `visit`. But if we can remove it then even better 👍